src/MenuButton.js

```diff
@@ -21,8 +23,12 @@ class MenuButton extends Component {
 
   handleFileInput (file) {
     const reader = new window.FileReader()
-    reader.onload = (event) => {
-      this.props.onClick(JSON.parse(event.target.result))
+    if (file.name.split('.').pop().toLowerCase() === 'json' || 'csv') {
```

Comment:
using `load_json_or_csv` means you don't need to check the filename. If a user entered a `.txt` file that happened to be comma separated, the code would work just fine.